moped-editor/src/views/projects/newProjectView/NewProjectMapBaseMap.js

```diff
@@ -0,0 +1,144 @@
+    width: "100%",
+    top: "0",
+  },
+}));
+
+/**
+ * Generates the basemap selector using the SpeedDial component
+ * @param {Object} containerRef - The reference to the container where this component will be rendered
+ * @param {function} handleBasemapChange - The function we will call with to change the basemap
+ * @param {string} mapStyle - The name of the current basemap
+ * @return {React.ReactPortal|null}
+ * @constructor
+ */
+const NewProjectMapBaseMap = ({
+  containerRef,
+  handleBasemapChange,
+  mapStyle,
+}) => {
+  const classes = useStyles();
+
+  /**
+   * Basemap Speed Dial State
+   */
+  const [mapBasemapSpeedDialOpen, setBasemapSpeedDialOpen] = useState(false);
+
+  /**
+   * Changes the current basemap and closes the speed dial menu
+   * @param basemapName
+   */
+  const handleBasemapSpeedDialClose = basemapName => {
+    setBasemapSpeedDialOpen(false);
+    if (basemapName) handleBasemapChange(basemapName);
+  };
+
+  /**
+   * Opens the speed dial menu
+   */
+  const handleBasemapSpeedDialOpen = () => {
+    setBasemapSpeedDialOpen(true);
+  };
+
+  return containerRef && containerRef.current
+    ? ReactDOM.createPortal(
+        <SpeedDial
+          ariaLabel="Basemap Select"
+          hidden={false}
+          icon={
+            <Typography
+              className={
+                mapBasemapSpeedDialOpen
+                  ? classes.mapStyleToggleTypographyOpen
+                  : classes.mapStyleToggleTypography
+              }
+            >
+              <Icon className={classes.mapStyleToggleLabelIcon}>layers</Icon>
+              <span className={classes.mapStyleToggleLabel}>Map</span>
+            </Typography>
+          }
+          onClose={() => handleBasemapSpeedDialClose(null)}
+          onOpen={handleBasemapSpeedDialOpen}
+          open={mapBasemapSpeedDialOpen}
+          direction={"left"}
+          FabProps={{
+            className:
+              mapStyle !== "streets"
+                ? classes.speedDialStreets
+                : classes.speedDialAerial,
+          }}
+        >
+          <SpeedDialAction
+            key={"streets"}
+            icon={
+              <Typography>
+                <span className={classes.mapStyleToggleLabel}>Streets</span>
+              </Typography>
+            }
+            tooltipTitle={"Streets Base Map"}
+            tooltipPlacement={"top"}
+            onClick={() => handleBasemapSpeedDialClose("streets")}
+          />
+          <SpeedDialAction
+            key={"aerial"}
+            icon={
+              <Typography>
+                <span className={classes.mapStyleToggleLabel}>Aerial</span>
+              </Typography>
+            }
+            tooltipTitle={"Aerial Base Map"}
+            tooltipPlacement={"top"}
+            onClick={() => handleBasemapSpeedDialClose("aerial")}
+          />
+        </SpeedDial>,
+        containerRef.current
+      )
+    : null;
+};
+
+export default NewProjectMapBaseMap;
```

Comment:
This entire file is a new base map component that uses Material's Speed-dial component to switch between aerial or street view.